packages/opentelemetry-basic-tracer/src/BasicTracer.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@opentelemetry/types';
+import { ScopeManager } from '@opentelemetry/scope-base';
+import {
+  BinaryTraceContext,
+  HttpTraceContext,
+  NoopSpan,
+  randomSpanId,
+  randomTraceId,
+  INVALID_SPAN_CONTEXT,
+  NEVER_SAMPLER,
+} from '@opentelemetry/core';
+import { BasicTracerConfig } from '../src/types';
+import { BinaryFormat, HttpTextFormat } from '@opentelemetry/types';
+
+/**
+ * This class represents a basic tracer.
+ */
+export class BasicTracer implements types.Tracer {
+  static defaultSpan = new NoopSpan(INVALID_SPAN_CONTEXT);
+
+  private _defaultAttributes: types.Attributes;
+  private _binaryFormat: types.BinaryFormat;
+  private _httpTextFormat: types.HttpTextFormat;
+  private _sampler: types.Sampler;
+  private _scopeManager: ScopeManager;
+
+  /**
+   * Constructs a new Tracer instance.
+   */
+  constructor(config: BasicTracerConfig) {
+    this._binaryFormat = config.binaryFormat || new BinaryTraceContext();
+    this._defaultAttributes = config.defaultAttributes || {};
+    this._httpTextFormat = config.httpTextFormat || new HttpTraceContext();
+    this._sampler = config.sampler || NEVER_SAMPLER;
```

Comment:
I think the expected default would be `ALWAYS_SAMPLED`. Is this enforced to be "never" in the spec?

Reply:
Related issue https://github.com/open-telemetry/opentelemetry-java/issues/347, **Decision: Default will be 100% sampling.**

Reply:
I changed it to `ALWAYS_SAMPLER1